Fix bug with routing

davidyell opened the pull request:
If the toolbar was loaded inside a prefix, the links would automatically take the prefix causing a 404.

Example, `http://example.com/admin/clear_cache/clear_cache/files`

By updating the `$linkUrl` on it's first use to remove the prefix configured in `core.php` the links can find the correct routes.

Commit:
Forcibly remove prefixes to correct incorrect routing.

ceeram commented on View/Elements/clear_cache_panel.ctp:
```diff
@@ -22,6 +22,11 @@
 		echo __d('clear_cache', 'Files') . ': ';
 		$linkUrl['action'] = 'files';
+
+		foreach (Configure::read('Routing.prefixes') as $prefix) {
```
putting this inside this else, the engines and group links still can be wrong.

davidyell replied:
Yeah, I thought that, but because the variable is not reset in any way, it affects all the following uses, meaning less code repetition. If we wanted to add it to all the links, it'd probably want a helper in there for generating the links.

ceeram replied:
not     `if (empty($content['folders'])) {`
Then it won't set it for the other `$content` keys
I would prefer doing this right after defining $linkUrl, no code repetition there too, but then you know for sure all links always get the prefix

davidyell replied:
Cool, I'll revisit the code and update the PR when I get some time :)

davidyell committed:
Refactored the repeated links into a helper. Shoehorned the helper into the controller using the panels beforeRender. Updated the panel to use the new controller

davidyell commented:
I've had time to take a proper stab at this. Hope you don't mind the refactoring. It's just the code felt a little repeated to me :)

davidyell commented:
Is there anything still outstanding I need to do on this PR?

ceeram commented:
Im not fond of the additional helper. The original PR was fine by itself except for the comment i left

Commit:
Updated the panel to remove the helper and push the prefix stripping out of the condition.

Commit:
Merge branch 'master' of https://github.com/davidyell/clear_cache

Conflicts:
	View/Elements/clear_cache_panel.ctp

davidyell commented:
Hopefully this update is okay. I've removed both the Component and the helper. I've also moved the strip of prefixes up into the main linkUrl section, so it will always affect all files, groups and engines.

Commit:
Merge branch 'master' of https://github.com/ceeram/clear_cache

Conflicts:
	View/Elements/clear_cache_panel.ctp

davidyell commented:
Fixed up the merge conflicts.

ceeram commented:
Thanks
